### PR TITLE
Organize code into a resource module

### PR DESCRIPTION
**Pull request by leplatrem**
* [x] Group resource, viewset, schema into a `resource` module. Since all this is optional to use.
* [x] Rename `Collection` to `Model`, since it is what it is :)
* [x] Document
* [x] Support deprecation

@Natim @ametaireau  thoughts ?

### CODE REVIEW

File: cliquet/resource/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from .base import *
```
Comment (Natim) [inline cliquet/resource/__init__.py]: We should not import * there, either move the base code here or define the ``__all__``  variable.

Comment (leplatrem) [inline cliquet/resource/__init__.py]: yes ok, but what do you think of the overall move ?

**Comment by leplatrem** [comment]
Ready to review !

**Comment by Natim** [comment]
It looks good to me.

**Pull request merged**